src/components/baseComponent.ts

```diff
@@ -141,4 +182,26 @@ export abstract class MgtBaseComponent extends LitElement {
     });
     this.dispatchEvent(event);
   }
+
+  /**
+   * Force reload the state.
+   * Use reload instead of load to ensure loading events are fired.
+   *
+   * @protected
+   * @returns {Promise<void>}
+   * @memberof MgtBaseComponent
+   */
+  protected async reload(): Promise<void> {
+    if (this._loading) {
```

Comment:
Just thinking through this, does it make sense to have a queue or a reloadRequired flag to ensure a reload request is not missed? 

The edge case here would be if a property (that requires the state to reload) gets updated while the state is loading, the reload will get ignored

Reply:
Good points. I don't really like the idea of a queue, because I see that getting messy without good cancellation logic. But I like the `reloadRequired`/`force` concept. I'll add that in.

Perhaps `reload()` should just return false if it is already loading. That, plus the `force` parameter should be enough imo.

What do you think?

Reply:
I agree a queue is unnecessary here since you will most likely need to reload once per multiple changes. reload could just call itself again if a reload was requested while it was loading.